app/controllers/admin/product_import_controller.rb

```diff
@@ -0,0 +1,69 @@
+require 'roo'
+
+class Admin::ProductImportController < Spree::Admin::BaseController
+
+  def import
+    unless params[:file]
+      redirect_to '/admin/product_import', :notice => 'File not found or could not be opened'
+      return
+    end
+
+    # Save uploaded file to tmp directory
+    upload = params[:file]
+    filename = Time.now.strftime('%d-%m-%Y-%H-%M-%S')+'.'+upload.original_filename.split('.').last
+    @filepath = File.open(Rails.root.join('tmp', filename), 'wb') do |f|
+      f.write(upload.read)
+      f.path
+    end
+
+    importer = ProductImporter.new(File.new(@filepath), editable_enterprises)
+
+    @valid_entries = importer.valid_entries
+    @invalid_entries = importer.invalid_entries
+
+    @valid_count = importer.valid_count
+    @invalid_count = importer.invalid_count
+    @has_valid_entries = importer.has_valid_entries?
+
+    @products_to_create = importer.products_create_count
+    @products_to_update = importer.products_update_count
```

Comment:
I had these in for debugging purposes whilst sorting out the model and controller, but yes, one of the next jobs is to remake all the views with a much better UI.